test/integration-test-main.sh

```diff
@@ -2243,59 +2251,58 @@ function test_not_existed_dir_obj() {
     echo data2 | aws_cli s3 cp - "s3://${TEST_BUCKET_1}/${OBJECT_NAME_2}"
 
     # Top directory
-    # shellcheck disable=SC2010
-    if ! ls -1 | grep -q '^not_existed_dir_single$'; then
-    echo "Expect to find \"not_existed_dir_single\" directory, but it is not found"
-    return 1;
+    local _LISTUP_OBJNAMES; _LISTUP_OBJNAMES="$(ls)"
```

Comment:
Is this a different change?  `ls -1` is not meaningful in non-interactive scripts.

Would `find -mindepth 1` address this more cleanly?  Happy to accept this as a separate PR.

Reply:
First, I will replace all occurrences of `ls -1` with `ls`.

And, previously I used the results of a one-time `ls` command to check for `not_existed_dir_single` and `not_existed_dir_parent` directories.
but I will fix this by calling an `ls` command for each directory.

Since the code of this part to be PR will be significantly changed (deleted), I will mark this as solved for now.